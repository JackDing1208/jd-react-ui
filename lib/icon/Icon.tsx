import React from "react";
import "./style.scss"
import classes from "../helper/className"
import "./importAllIcons"
// import "./svg/react.svg"

interface Props extends React.SVGAttributes<SVGElement>{   //继承svg标签自带属性
  name: string,
  size?: string
}


const Icon: React.FunctionComponent<Props> = (props) => {
  const {name,className,size,...rest}=props
  return (
      <svg
        className={classes("jd-icon",className)}
        style={{width:size,height:size}}
        {...rest}>
        <use xlinkHref={`#${name}`}/>
      </svg>
  );
};


export default Icon;
