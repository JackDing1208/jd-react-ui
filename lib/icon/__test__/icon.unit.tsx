import * as renderer from 'react-test-renderer'
import * as React from 'react'
import Icon from '../Icon'

describe('Icon', () => {
  it('can render', () => {
    const json = renderer.create(<Icon name={"react"}/>).toJSON()
    expect(json).toMatchSnapshot()
  })
})
