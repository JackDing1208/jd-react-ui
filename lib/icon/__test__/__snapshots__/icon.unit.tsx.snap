// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon can render 1`] = `
<svg
  className="jd-icon"
  style={
    Object {
      "fill": undefined,
      "height": undefined,
      "width": undefined,
    }
  }
>
  <use
    xlinkHref="#react"
  />
</svg>
`;
